### PR TITLE
Fix `logabsbeta` for BigFloat

### PR DESCRIPTION
**Pull request by devmotion**
This PR adds a fallback definition for `_loggammadiv`. This fixes test errors in https://github.com/TuringLang/DistributionsAD.jl/pull/172.

**Motivation:**
Version 1.4.1, or more concretely https://github.com/JuliaMath/SpecialFunctions.jl/pull/316 (@andreasnoack), caused some regressions of `logbeta` and `logabsbeta` with `BigFloat`s.

On 1.4.0:
```julia
julia> using SpecialFunctions

julia> beta(big(nextfloat(0.0)), big(nextfloat(8.0)))
2.024022533073106183524953467189173070495566497641421183569013580274303395679953e+323

julia> logbeta(big(nextfloat(0.0)), big(nextfloat(8.0)))
744.4400719213812623141072984460816341130871443029141429256103301959047499954517

julia> logabsbeta(big(nextfloat(0.0)), big(nextfloat(8.0)))
(744.4400719213812623141072984460816341130871443029141429256103301959047499954517, 1)
```

On 1.4.1:
```julia
julia> beta(big(nextfloat(0.0)), big(nextfloat(8.0)))
2.024022533073106183524953467189173070495566497641421183569013580274303395679953e+323

julia> logbeta(big(nextfloat(0.0)), big(nextfloat(8.0)))
ERROR: MethodError: no method matching _loggammadiv(::BigFloat, ::BigFloat)
Stacktrace:
 [1] loggammadiv(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/W3J2O/src/beta_inc.jl:12
 [2] logabsbeta(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/W3J2O/src/gamma.jl:822
 [3] logbeta(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/W3J2O/src/gamma.jl:788
 [4] top-level scope
   @ REPL[15]:1

julia> logabsbeta(big(nextfloat(0.0)), big(nextfloat(8.0)))
ERROR: MethodError: no method matching _loggammadiv(::BigFloat, ::BigFloat)
Stacktrace:
 [1] loggammadiv(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/W3J2O/src/beta_inc.jl:12
 [2] logabsbeta(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/W3J2O/src/gamma.jl:822
 [3] top-level scope
   @ REPL[16]:1
```

`beta` still works since it uses the definition in mpfr on my machine. Actually, the problem was present in SpecialFunctions 1.4.0 as well but it is easier to trigger it in 1.4.1 since the branch is chosen not only for inputs with large difference in magnitude anymore (on 1.4.0: https://github.com/JuliaMath/SpecialFunctions.jl/blob/feccbf44f8a8caed9a8eb85d4025c8859b9ab537/src/gamma.jl#L842, on 1.4.1: https://github.com/JuliaMath/SpecialFunctions.jl/blob/dd414811d6eb3538d4e1b5b0f42f8f15d8ec77a0/src/gamma.jl#L821).

On 1.4.0:
```julia
julia> logabsbeta(big(nextfloat(1e5)), big(1.0))
ERROR: MethodError: no method matching _loggammadiv(::BigFloat, ::BigFloat)
Stacktrace:
 [1] loggammadiv(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/L13jJ/src/beta_inc.jl:12
 [2] logabsbeta(a::BigFloat, b::BigFloat)
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/L13jJ/src/gamma.jl:843
 [3] top-level scope
   @ REPL[14]:1
```

With this PR all examples work:
```julia
julia> beta(big(nextfloat(0.0)), big(nextfloat(8.0)))
2.024022533073106183524953467189173070495566497641421183569013580274303395679953e+323

julia> logbeta(big(nextfloat(0.0)), big(nextfloat(8.0)))
744.4400719213812623141072984460816341130871443029141429256103301959047499954517

julia> logabsbeta(big(nextfloat(0.0)), big(nextfloat(8.0)))
(744.4400719213812623141072984460816341130871443029141429256103301959047499954517, 1)

julia> logabsbeta(big(nextfloat(1e5)), big(1.0))
(-11.51292546497022856560910955709032851649991676439065368877238753498953004770512, 1)
```

### CODE REVIEW

**Comment by devmotion** [comment]
Test failures are the same as on the master branch, it seems.

**Comment by giordano** [comment]
> Test failures are the same as on the master branch, it seems.

I'm lost, are they due to #319?  Is there anything we can do to have a usable state?

**Comment by devmotion** [comment]
Hopefully https://github.com/JuliaMath/SpecialFunctions.jl/pull/327 fixes it.

File: src/beta_inc.jl [hunk 1]
```diff
@@ -9,8 +9,9 @@ const exparg_p =  log(prevfloat(floatmax(Float64)))
 
 Computes ``log(\\Gamma(b)/\\Gamma(a+b))`` when b >= 8
 """
-loggammadiv(a::Number, b::Number) = _loggammadiv(float(a), float(b))
+loggammadiv(a::Number, b::Number) = _loggammadiv(promote(float(a), float(b))...)
 
+_loggammadiv(a::Number, b::Number) = loggamma(b) - loggamma(a + b)
```
Comment (andreasnoack) [inline src/beta_inc.jl:14]: I think it would be useful to add a comment here as a reminder that a proper implementation should be added at some point.

Comment (devmotion) [reply src/beta_inc.jl]: I added a comment.

**Review by andreasnoack — APPROVED** [review]

**Pull request merged**